### PR TITLE
Add docs and allow for multiple precisions

### PR DESCRIPTION
**Pull request by martindurant**
Starting the process. I have not touched the tests, but appears to work with float64.

### CODE REVIEW

File: numcodecs/bitround.py [hunk 1]
```diff
@@ -4,32 +4,68 @@
...
     def encode(self, buf):
-        if self.keepbits == 23:
-            return buf
-        # TODO: figure out if we need to make a copy
-        # Currently this appears to be overwriting the input buffer
-        # Is that the right behavior?
-        a = ensure_ndarray(buf).view()
-        assert a.dtype == np.float32
-        b = a.view(dtype=np.int32)
+        """Create int array by rounding floating data
```
Comment (rabernat) [inline numcodecs/bitround.py]:
```suggestion
        """Create int array by rounding floating-point data
```

**Comment by martindurant** [comment]
Assuming complex is simply pairs of float, it might be reasonable to apply to that type too.

File: numcodecs/bitround.py [hunk 2]
```diff
@@ -4,32 +4,68 @@
...
+}
+inverse = {
+    "int16": np.float16,
+    "int32": np.float32,
+    "int64": np.float64
+}
+
 
 class BitRound(Codec):
+    """Real information content algorithm
```
Comment (rabernat) [inline numcodecs/bitround.py]:
```suggestion
    """Floating-point bit rounding codec
```

File: numcodecs/bitround.py [hunk 3]
```diff
@@ -4,32 +4,68 @@
...
+
+    Drops data in the least significant part of the floating point mantissa,
```
Comment (rabernat) [inline numcodecs/bitround.py]:
```suggestion
    Drops a specified number of bits from the floating point mantissa,
```

File: numcodecs/bitround.py [hunk 4]
```diff
@@ -4,32 +4,68 @@
...
+    leaving an array uch more amenable to compression. See
```
Comment (rabernat) [inline numcodecs/bitround.py]:
```suggestion
    leaving an array more amenable to compression. The number of bits to keep should
    be determined by a information analysis of the data to be compressed. See 
```

**Review by rabernat — COMMENTED** [review]
Important to clarify that this codec does not actually determine the number of significant bits. It just applies a specified bit rounding.

**Pull request merged**